astroid/inference.py

```diff
@@ -759,7 +763,7 @@ def _get_binop_contexts(context, left, right):
         yield new_context
 
 
-def _same_type(type1, type2):
+def _same_type(type1, type2) -> bool:
```

Comment:
This is interesting as adding annotations to `type1` will cause a type error... Hmm, we should probably investigate this.

Reply:
This function is used twice. In both cases, the inputs can be `InferenceResult | None`, but this function assumes that it will be `InferenceResult`.

Reply:
`Proxy` is involved too. Messy!

Reply:
😅